### PR TITLE
Update cache key generation to use backend compiler name too

### PR DESCRIPTION
**Pull request by MarkMcCaskey**

### CODE REVIEW

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

File: lib/runtime-core/src/cache.rs [hunk 1]
```diff
@@ -41,12 +42,13 @@ impl WasmHash {
     /// # Note:
     /// This does no verification that the supplied data
     /// is, in fact, a wasm module.
-    pub fn generate(wasm: &[u8]) -> Self {
+    pub fn generate(wasm: &[u8], backend: Backend) -> Self {
```
Comment (syrusakbary) [inline lib/runtime-core/src/cache.rs]: Can we have this as a non-breaking change. I think near was relying on `generate` and I would like to keep the function signature the same.

Perhaps something like?
```
    pub fn generate_for_backend(wasm: &[u8], backend: Option<Backend>) -> Self {
    }
    pub fn generate(wasm: &[u8]) -> Self {
       Self::generate_for_backend(wasm, Some(Cranelift);
    }
```

Comment (MarkMcCaskey) [reply lib/runtime-core/src/cache.rs]: Sure, the main reason I didn't do that is that it has the potential for bad things to happen, but I'll call it out in a doc comment and call it good enough!

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors retry

**Comment by MarkMcCaskey** [comment]
bors retry

**Pull request merged**